src/Mod/Import/App/ImportOCAF2.cpp

```diff
@@ -305,22 +305,7 @@ App::DocumentObject *ImportOCAF2::expandShape(
     if(shape.IsNull() || !TopExp_Explorer(shape,TopAbs_VERTEX).More())
         return 0;
 
-    // When saved as compound, STEP file does not support instance sharing,
-    // meaning that even if the source compound may contain child shapes of
-    // shared instances, or multiple hierarchies, those information are lost
-    // when saved to STEP, everything become flat and duplicated. So the code
-    // below is not necessary.
-#if 0
-    auto baseShape = shape.Located(TopLoc_Location());
-    auto it = myShapes.find(baseShape);
-    if(it!=myShapes.end()) {
-        auto link = static_cast<App::Link*>(doc->addObject("App::Link","Link"));
-        link->Visibility.setValue(false);
-        link->setLink(-1,it->second.obj);
-        setPlacement(&link->Placement,shape);
-        return link;
-    }
-#endif
```

Comment:
The comment is my understanding by testing and browsing through OCC code, which is kind of a surprise to me. I am not sure if OCC will change this behavior in the future. I think it worth keeping the comment and code here.